models/users.js

```diff
@@ -178,26 +179,38 @@ const fetchPaginatedUsers = async (query) => {
   }
 };
 
-const fetchAllUsers = async () => {
+const fetchFilteredUsers = async (usernames = []) => {
```

Comment:
I feel this function name should just be `fetchUsers(providedUsernames)`

Why should this function be aware that the list was 'filtered' down somewhere else?

It's contract is simple: Give it a list of usernames, and it will give you back the data for all those usernames.